src/assessments/seo/keyphraseLengthAssessment.js

```diff
@@ -1,31 +1,116 @@
+				requiresLengthAndMax: false,
+			},
+			okay: {
+				score: 6,
+				resultText: "Your keyphrase is %1$d words long. That's more than the recommended maximum of %2$d words. " +
+				"You might want to make the keyphrase a bit shorter.",
+				requiresLengthAndMax: true,
+			},
+			bad: {
+				score: 3,
+				resultText: "Your keyphrase is %1$d words long. That's way more than the recommended maximum of %2$d " +
+				"words. Make the keyphrase shorter.",
+				requiresLengthAndMax: true,
+			},
+			veryBad: {
+				score: -999,
+				resultText: "No focus keyword was set for this page. If you do not set a focus keyword, no score can " +
+				"be calculated.",
+				requiresLengthAndMax: false,
+			},
+		};
+
+		this.identifier = "keyphraseLength";
+		this._config = merge( defaultConfig, config );
+	}
+
+	/**
+	 * Assesses the keyphrase presence and length.
+	 *
+	 * @param {Paper} paper The paper to use for the assessment.
+	 * @param {Researcher} researcher The researcher used for calling research.
+	 * @param {Jed} i18n The object used for translations.
+	 *
+	 * @returns {AssessmentResult} The result of this assessment.
+	 */
+	getResult( paper, researcher, i18n ) {
+		this._keyphraseLength = researcher.getResearch( "keyphraseLength" );
+		let assessmentResult =  new AssessmentResult();
+
+		const calculatedScore = this.calculateScore();
+		assessmentResult.setScore( calculatedScore.score );
+		assessmentResult.setText( this.translateScore( calculatedScore.resultText, calculatedScore.requiresLengthAndMax, i18n ) );
+
+		return assessmentResult;
 	}
 
-	return assessmentResult;
+	/**
+	 * Calculates the score result based on the keyphraseLength research.
+	 *
+	 * @returns {Object} object with score and text.
+	 */
+	calculateScore() {
+		if ( this._keyphraseLength < this._config.parameters.recommendedMinimum ) {
+			return this._config.veryBad;
+		}
+
+		if ( inRange( this._keyphraseLength, this._config.parameters.recommendedMinimum, this._config.parameters.recommendedMaximum + 1 ) ) {
+			return this._config.good;
+		}
+
+		if ( inRange( this._keyphraseLength, this._config.parameters.recommendedMaximum + 1, this._config.parameters.acceptableMaximum + 1 ) ) {
+			return this._config.okay;
+		}
+
+		if ( this._keyphraseLength > this._config.parameters.acceptableMaximum ) {
+			return this._config.bad;
+		}
+	}
+
+	/**
+	 * Translates the score result based into specific feedback to the user.
+	 *
+	 * @param {text} resultText The text of feedback for a given range of values of the assessment result.
+	 * @param {boolean} requiresLengthAndMax Whether feedback needs to include keyphraseLength and recommendedMaximum.
+	 * @param {object} i18n The i18n-object used for parsing translations.
+	 *
+	 * @returns {string} Text feedback.
+	 */
+	translateScore( resultText, requiresLengthAndMax, i18n ) {
+		if ( requiresLengthAndMax ) {
+			return i18n.sprintf( i18n.dgettext( "js-text-analysis", resultText ), this._keyphraseLength,
```

Comment:
Translator comments should be included for `keyphraseLength` and `recommendedMaximum`, see other assessments with variables.